quirc/examples/esp-eye-qrcode/main/app_main.c

```diff
@@ -0,0 +1,85 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include "esp_log.h"
+#include "esp_system.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "app_peripherals.h"
+#include "quirc.h"
+
+static const char *TAG = "APP_CODE_SCANNER";
+
+static void decode_task()
+{
+    if(ESP_OK != app_camera_init()) {
+        vTaskDelete(NULL);
+        return;
+    }
+
+    camera_fb_t *fb = NULL;
+
+    // Initializing the quirc handle
+    struct quirc *q = quirc_new();
+    if (!q) {
+        ESP_LOGE(TAG,"Failed to allocate memory\n");
+        exit(1);
+    }
+
+
+    // Get image size through fb parameters
+    fb = esp_camera_fb_get();
+    if(fb == NULL){
+        ESP_LOGI(TAG, "camera get failed\n");
+    }
+
+    uint16_t p_width = fb->width;
+    uint16_t p_height = fb->height;
+
+
+    if (quirc_resize(q, p_width,p_height) < 0) {
+        ESP_LOGE(TAG,"Failed to allocate video  memory\n");
+        exit(1);
+    }   
+
+    struct quirc_code code;
+    struct quirc_data data;
+    quirc_decode_error_t err;
+    uint16_t num_codes;
+
+    while (1)
+    {
+        fb = esp_camera_fb_get();
+        if(fb == NULL){
+            ESP_LOGI(TAG, "camera get failed\n");
+            continue;
+        }
+        // Decode Progress
+    
+        memcpy(quirc_begin(q, NULL, NULL), fb->buf, fb->len);
+        quirc_end(q);
+
+        num_codes = quirc_count(q);
+        for (uint16_t i = 0; i < num_codes; i++) {
+
+            quirc_extract(q, i, &code);
+            /* Decoding stage */
+            err = quirc_decode(&code, &data);
+            if (err)
+                printf("%d/%d] DECODE FAILED: %s\n", i+1,num_codes,quirc_strerror(err));
+            else
+                printf("%d/%d] DATA: %s\n",i+1,num_codes,data.payload);
+        }
+
+        esp_camera_fb_return(fb);
+        vTaskDelay(10 / portTICK_PERIOD_MS);
+    }
+
+    quirc_destroy(q);
+}
```

Comment:
`vTaskDelete(NULL);` missing here?

Reply:
This part of the code isn't supposed to be reached. I'm not sure what the best approach would be here. 
Do you suggest the vTaskDelete or to remove the quirc_destroy completely?

Reply:
Depends on the error handling strategy — see the other comment about this.

If you decide to handle errors by terminating the task, probably having the cleanup path with quirc_destroy would make sense. You could use an `out:` goto label and use macros like `ESP_GOTO_ON_ERROR` to handle errors and perform cleanup.

If you decide to handle errors by aborting (which is not an unreasonable strategy for an example project) then quirc_destroy can be removed since it's unreachable.

Reply:
Ok then since as you say it's an example, I think it's best to keep it the simplest possible. I removed the quirc_destroy.